ui/raidboss/data/06-ew/raid/p6s.ts

```diff
@@ -190,6 +200,143 @@ const triggerSet: TriggerSet<Data> = {
       netRegex: NetRegexes.ability({ id: '788B', source: 'Hegemone', capture: false }),
       response: Responses.moveAway(),
     },
+    {
+      id: 'P6S Predation Debuff Collect',
+      // CF7 Glossal Resistance Down (Snake Icon)
+      // CF8 Chelic Resistance Down (Wing Icon)
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: ['CF7', 'CF8'] }),
+      condition: Conditions.targetIsYou(),
+      run: (data, matches) => data.predationDebuff = matches.effectId,
+    },
+    {
+      id: 'P6S Predation Bait Order',
+      // Using Aetheronecrosis (CF9)
+      // These come out as 20s, 16s, 12s, or 8s
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: 'CF9' }),
+      condition: Conditions.targetIsYou(),
+      preRun: (data, matches) => data.aetheronecrosisDuration = parseFloat(matches.duration),
+      delaySeconds: 0.1,
+      durationSeconds: (_data, matches) => {
+        const duration = parseFloat(matches.duration);
+        // First Dual Predation is 3.7s before expiration
+        // Remaining Dual Predations are 12.3s (second), 12.4s (third/fourth)
+        return duration === 20 ? duration - 3.8 : duration + 12.3;
+      },
+      infoText: (data, matches, output) => {
+        const AetheronecrosisMap: { [duration: number]: string } = {
+          20.00: output.firstBait!(),
```

Comment:
This is not safe, sometimes log lines get written out slightly late and the effect has already started, e.g. this could be `19.98` or something and not `20.00`.  I think you need to do something more like https://github.com/quisquous/cactbot/blob/main/ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.ts#L701-L716 where you do a more rough comparison against times.  Sorry to make it more complicated, just trying to safeguard against issues I've seen in the past.

Reply:
Thanks for the heads up. I've modified it to compare to <= duration and > 16 where 20 is used.